File: Data Structures & Algorithms/concatenation-of-array/submission-1.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> result(nums.size() * 2);
        int i = 0;
        int n = nums.size();

        while (i < n) {
            result[i] = nums[i];
            result[i + n] = nums[i];
            ++i;
        }
        return result;
    }
};

File: Data Structures & Algorithms/count-number-of-islands/submission-1.java
class Solution {
    public void dfs(char[][] grid, int r, int c) {
        if (r < 0 || r >= grid.length) return;
        if (c < 0 || c >= grid[0].length) return;
        if (grid[r][c] == '0') return;

        grid[r][c] = '0';

        dfs(grid, r - 1, c);
        dfs(grid, r + 1, c);
        dfs(grid, r, c - 1);
        dfs(grid, r, c + 1);
    }

    public int numIslands(char[][] grid) {
        if (grid.length == 0) return 0;

        int m = grid.length, n = grid[0].length;
        int result = 0;

        for (int r = 0; r < m; ++r) {
            for (int c = 0; c < n; ++c) {
                if (grid[r][c] == '1') {
                    dfs(grid, r, c);
                    ++result;
                }
            }
        }
        return result;
    }
}
